main/src/main/scala/sbt/Keys.scala

```diff
@@ -137,6 +137,7 @@ object Keys {
   val sbtPlugin = SettingKey[Boolean]("sbt-plugin", "If true, enables adding sbt as a dependency and auto-generation of the plugin descriptor file.", BMinusSetting)
   val printWarnings = TaskKey[Unit]("print-warnings", "Shows warnings from compilation, including ones that weren't printed initially.", BPlusTask)
   val fileInputOptions = SettingKey[Seq[String]]("file-input-options", "Options that take file input, which may invalidate the cache.", CSetting)
+  val scalaCompilerBridgeSource = SettingKey[ModuleID]("scala-compiler-bridge-source", "Configures the module ID of the sources of the compiler bridge.", CSetting)
```

Comment:
Actually the idea is to replace the hierarchy by specifying directly in the build definition what version of the compiler bridge is required. After the compiler bridges have been split, it will become something like:

```scala
scalaCompilerBirdgeSource := scalaVersion.value match {
  case v if v startsWith "2.11" => "org.scala-sbt" % "compiler-interface" % sbtVersion.value
  case _  => "org.scala-sbt" % "compiler-interface-old" % sbtVersion.value
}
```

This way we keep the flexibility, but we don't rely on the hypothetical existence of some artefacts. The same build could yield totally different result depending on the availability of a specific compiler interface source at some point.